docs/new_agents.rst

```diff
@@ -1,3 +1,390 @@
+                           ['--time-per-file', '3600'],
+                           ['--data-dir', '/data/']]},
+
+            # Arduino
+            {'agent-class': 'ArduinoAgent',
+             'instance-id': 'arduino',
+             'arguments': []},
+        ]
+    }
+
+When adding a new Agent, the ``'agent-class'`` entry should match the name of
+your class in the Agent file. The ``'arguments'`` entry should match any
+arguments that you added to ``pgroup`` at the end of your Agent file.
+
+In this example, the ``'agent-instances'`` are found under a host called 
+``grumpy``, which in this case is the name of the host computer. However, when 
+writing an Agent that will be broadly useful, we may choose to Dockerize the 
+Agent (and its dependencies). For more on this, see the Docker section of this 
+documentation.
+
+
+Docker
+------
+A Docker container creates a virtual environment in which you can package 
+applications with their libraries and dependencies. OCS is sometimes installed 
+in a Docker container (for ease of installation). For Agents that are not 
+meant solely to be used with one lab computer, it can be useful to add them to a 
+Docker container as well. This requires creating a ``Dockerfile`` for your Agent 
+and adding the Agent capabilites to your OCS Docker container in a 
+``docker-compose.yml`` file. Adding your Agent in the ``docker-compose.yml`` file 
+will also allow you to view your data feed when you run the Agent.
+
+To create a ``Dockerfile``, change directories to the directory containing your 
+Agent file. Within this directory, create a file called ``Dockerfile``. The format 
+of this file is as follows (using the Arduino as an example):
+
+::
+
+        # SOCS Arduino Agent
+        # socs Agent container for interacting with an Arduino
+
+        # Use socs base image
+        FROM socs:latest
+
+        # Set the working directory to registry directory
+        WORKDIR /app/agents/arduino/
+
+        # Copy this agent into the app/agents directory
+        COPY . /app/agents/arduino/
+
+        # Run registry on container startup
+        ENTRYPOINT ["python3", "-u", "arduino_agent.py"]
+
+
+In this case, the ``WORKDIR``, ``COPY``, and ``ENTRYPOINT`` arguments are all set 
+specifically to the correct directories and files for the Arduino agent. You can 
+additionally connect the container to a Crossbar (WAMP) server; see the Sisock 
+documentation for more on this. 
+
+To include your new Agent among the services provided in your OCS Docker 
+container, navigate to the ``docker-compose.yml`` file in the same sub-directory 
+as your ``default.yaml`` or ``your_institution.yaml`` file. Within 
+``docker-compose.yml``, you should find (or create) a list of services that the 
+docker container provides. You can add your new agent following the example format:
+
+::
+
+  services:
+    arduino:
+      image: grumpy.physics.yale.edu/sisock-data-feed-server:v0.2.12-1-g52852b4
```

Comment:
This is only adding the sisock 'data server' which allows you to view a data feed from your agent. This doesn't add the Agent container itself. Certainly this container configuration is needed if you want to view your data, but once you've built your own image you need to add that as well. For instance, on grumpy, we have the data feed servers for the 372, but also the agents themselves:
```
  ocs-LSA22YE:
    image: grumpy.physics.yale.edu/ocs-lakeshore372-agent:79460f9
    hostname: grumpy-docker
    network_mode: "host"
    volumes:
      - ${OCS_CONFIG_DIR}:/config:ro
    command:
      - "--instance-id=LSA22YE"
```

For the 240's (which adds a device to the container, something you'll also need to do for the Arduino Agent):
```
  ocs-LSA24R5:
    image: grumpy.physics.yale.edu/ocs-lakeshore240-agent:79460f9
    depends_on:
      - "sisock-crossbar"
    devices:
      - "/dev/LSA24R5:/dev/LSA24R5"
    hostname: nuc-docker
    volumes:
      - ${OCS_CONFIG_DIR}:/config:ro
    command:
      - "--instance-id=LSA24R5"
      - "--site-hub=ws://sisock-crossbar:8001/ws"
      - "--site-http=http://sisock-crossbar:8001/call"
```